src/http-request-handler.ts

```diff
@@ -243,18 +236,121 @@ export class HttpRequestHandler implements IHttpRequestHandler {
+            url,
+            method: type as Method,
+            [key]: data || {},
+        };
+    }
+
+    /**
+     * Process global Request Error
+     *
+     * @param {Error} error      Error instance
+     * @returns {AxiosInstance} Provider's instance
+     */
+    protected processRequestError(error: Error): void {
+        if (axios.isCancel(error)) {
+            return;
+        }
+
+        const errorHandler = new HttpRequestErrorHandler(
             this.logger,
-            this.httpRequestErrorService,
-            this.strategy
-        ).process(err);
+            this.httpRequestErrorService
+        );
+
+        errorHandler.process(error);
+    }
+
+    /**
+     * Output error response depending on chosen strategy
+     *
+     * @param {Error} error      Error instance
+     * @param {EndpointConfig} requestConfig   Per endpoint request config
+     * @returns {AxiosInstance} Provider's instance
+     */
+    protected async outputErrorResponse(error: Error, requestConfig: EndpointConfig): Promise<IRequestResponse> {
+        const isRequestCancelled = requestConfig.cancelToken && axios.isCancel(error);
+        const errorHandlingStrategy = requestConfig.strategy || this.strategy;
+
+        // By default cancelled requests aren't rejected
+        if (isRequestCancelled && !requestConfig.rejectCancelled) {
+            return this.defaultResponse;
+        }
+
+        if (errorHandlingStrategy === 'silent') {
+            // Hang the promise
+            await new Promise(() => null);
+
+            return this.defaultResponse;
+        }
+
+        // Simply rejects a request promise
+        if (errorHandlingStrategy === 'reject' || errorHandlingStrategy === 'throwError') {
+            return Promise.reject(error);
+        }
+
+        return this.defaultResponse;
+    }
+
+    /**
+     * Output error response depending on chosen strategy
+     *
+     * @param {Error} error                     Error instance
+     * @param {EndpointConfig} requestConfig    Per endpoint request config
+     * @returns {*}                             Error response
+     */
+    public isRequestCancelled(error: Error, requestConfig: EndpointConfig): boolean {
+        return requestConfig.cancelToken && axios.isCancel(error);
+    }
+
+    /**
+     * Automatically Cancel Previous Requests
+     *
+     * @param {string} type                    Request type
+     * @param {string} url                     Request url
+     * @param {EndpointConfig} requestConfig   Per endpoint request config
+     * @returns {AxiosInstance} Provider's instance
+     */
+    protected addCancellationToken(type: string, url: string, requestConfig: EndpointConfig) {
+        // Both disabled
+        if (!this.cancellable && !requestConfig.cancellable) {
+            return {};
+        }
+
+        // Explicitly disabled per request
+        if (typeof requestConfig.cancellable !== "undefined" && !requestConfig.cancellable) {
```

Comment:
Upon u say !* means either `undefined`, `null`, `false`, technically is a repetitive condition check in this case.

You're saying this

```
if (this.cancellable is typeof value of  `undefined`, `null`, `false` && requestConfig as well `undefined`, `null`, `false`) >> return empty object 
``

This already repetitive, but not just, in case u gonna return a false can as well make multiple condition check
```
(!this.cancellable && !requestConfig.cancellable || typeof requestConfig.cancellable !== "undefined" && !requestConfig.cancellable) >> return
```

In this case can be simplified as firstly condition mentions the needs to return